dashboard/app/models/user.rb

```diff
@@ -456,6 +458,22 @@ def email_matches_for_oauth_upgrade
   validates_presence_of :email_preference_source, if: -> {email_preference_opt_in.present?}
   validates_presence_of :email_preference_form_kind, if: -> {email_preference_opt_in.present?}
 
+  # Validations for adding parent email notifications
+  before_validation :parent_email_preference_setup, if: :parent_email_preference_opt_in_required?
+  validates_inclusion_of :parent_email_preference_opt_in, in: %w(yes no), if: :parent_email_preference_opt_in_required?
+  validates_presence_of :parent_email_preference_email, if: :parent_email_preference_opt_in_required?
+  validates_presence_of :parent_email_preference_request_ip, if: :parent_email_preference_opt_in_required?
+  validates_presence_of :parent_email_preference_source, if: :parent_email_preference_opt_in_required?
+
+  def parent_email_preference_opt_in_required?
+    # This is a checkbox which either has the value '0' or '1'
```

Comment:
Is this a checkbox in the form?

Reply:
Yeah, on the account signup we have a checkbox saying "I'm a parent signing up my student". In the user.rb code, I'm using this as a flag to indicate we should be trying to do parent notification validations/updates.

Reply:
In your modal, you could just set a hidden field to have the value "1"